gui/src-tauri/src/main.rs

```diff
@@ -38,22 +39,53 @@ fn update_window_state(
 	Ok(())
 }
 
-fn main() {
+#[tauri::command]
+fn logging(msg: String) {
+	log::info!(target: "webview", "[webview] {}", msg)
+}
+
+#[tauri::command]
+fn erroring(msg: String) {
+	log::error!(target: "webview", "[webview] {}", msg)
+}
+
+#[tauri::command]
+fn warning(msg: String) {
+	log::warn!(target: "webview", "[webview] {}", msg)
+}
+
+fn main() -> Result<()> {
+	log_panics::init();
+	let hook = panic::take_hook();
 	// Make an error dialog box when panicking
-	panic::set_hook(Box::new(|panic_info| {
-		println!("{}", panic_info);
+	panic::set_hook(Box::new(move |panic_info| {
+		hook(panic_info);
```

Comment:
What happens if panic hook aborts or does something wacky? Presumably `show_error` should happen first?

Reply:
`show_error` sleeps until the user closes the prompt, what do you think we should do?

Reply:
I think we should let the panic handler block, because at least we guarantee that the user sees the popup.